src/Pages/SelectedCategory/SelectedCategory.jsx

```diff
@@ -1,47 +1,108 @@
+import { Header } from "../../components/Header/Header";
+import { Sidebar } from "../../components/Sidebar/Sidebar";
+import { Modal } from "../../components/Modal/Modal";
+import { addToHistory } from "../../Utility/add-to-history";
+import { useHistory } from "../../Context/history-context";
+import { usePlaylist } from "../../Context/platlist-context";
+import { useWatchLater } from "../../Context/watch-leter-context";
+import { deleteFromWatchLater } from "../../Utility/deleteFromWatchLater";
+import { addToWatchLater } from "../../Utility/addToWatchLater";
+import { useAuth } from "../../Context/auth-context";
 export const SelectedCategory = () => {
-  const {loader, setLoader} = useLoader()
-  const [selectedCategories, setSelectedCategory] = useState([]);
-  const { _id } = useParams();
-  // console.log(_id)
-  const getCategories = async () => {
-    try {
-      setLoader(true);
+  const { loader, setLoader } = useLoader();
+  const { setHistory } = useHistory();
+  const { playlistState, playlistDispatch } = usePlaylist();
+  const { watchLaterState, watchLaterDispatch } = useWatchLater();
+  const { watchedVideo } = watchLaterState;
+  const [filteredCategoies, setFilteredCategories] = useState([]);
+  const { categoryName } = useParams();
+  const { data, setData } = useData();
+  const { bg, modal } = playlistState;
+  const { auth } = useAuth();
+  const navigate = useNavigate();
+  useEffect(() => {
+    getVideos(setLoader, setData);
+  }, []);
 
-      const response = await axios.get(`/api/categories/${_id}`);
-      setLoader(false);
-      setSelectedCategory(response.data?.category);
-    } catch (error) {
-      console.log(error);
-    }
-  };
   useEffect(() => {
-    getCategories();
-  }, [_id]);
-  const { categoryName, categoryImage, description } = selectedCategories;
+    filterData();
+  }, [data]);
+
+  const filterData = () => {
+    const filteredData = data.filter(
+      (item) => item.categoryName === categoryName.toLowerCase()
+    );
+    setFilteredCategories(filteredData);
+  };
 
   return (
-    <>
-      {loader && <h1 style={{ textAlign: "center" }}>Loading...</h1>}
-      <div className="plr-1 category-container">
-        <div className="card shadow ">
-          <img className="card-img" src={categoryImage} alt={categoryName} />
-          <div>
-            <div className="card-body">
-              <h3 className="card-title ">{categoryName}</h3>
-              <small className="card-subtitle">{description}</small>
-            </div>
-            <button className="btn watch-now-btn mt-1">Watch Now</button>
-          </div>
+    <div style={{ backgroundColor: bg }}>
+      <Header />
+
+      <div className="grid-container">
+        <Sidebar />
+        <div className="postion-center">{modal && <Modal />}</div>
+        {loader && <h1 className="center">...Loading</h1>}
+        <div className="video-listing-main ">
+          {filteredCategoies.map((video) => {
```

Comment:
Suggestion : You can create another component of `FilteredCategoriesVideo` and pass the component in the map , and send mapped video as `props` to the component by this your code will look short and refactored

Reply:
yes I will do that